### PR TITLE
[automatic-failover] Support double-threshold logic with circuitbreaker

### PR DESCRIPTION
**Pull request by atakavci**
Double-threshold logic appears to be in place, with no additional development needed beyond the built-in functionality (as used during the POC).
This PR introduces tests covering the double-threshold logic and the full range of `CircuitBreaker` functionality, including edge cases and boundary conditions.
There is one minor improvement with `CircuitBreaker` by removing the unnecessary use of `Predicate`.

**UPDATE**
Based on reviews, here some additional changes;
- Add recordResult(Throwable), recordSuccess(), and recordFailure() public methods for recording outcomes
- Add getSnapshot() to expose metrics directly without exposing internal CircuitBreakerMetrics
- Change getMetrics() to package-private for internal use only
- Optimize evaluateMetrics() to avoid redundant snapshot calls
- Update endpoint implementations and integration tests to use new API

### CODE REVIEW

File: src/test/java/io/lettuce/core/failover/CircuitBreakerUnitTests.java [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+            circuitBreaker.evaluateMetrics();
+
+            // Then: circuit should open because failure count threshold is met (failureRateThreshold=0.0 means ignore
+            // percentage)
+            assertThat(circuitBreaker.getCurrentState()).isEqualTo(CircuitBreaker.State.OPEN);
+        }
+
+        @Test
+        @DisplayName("When failureRateThreshold=0.0, circuit opens as soon as minimum count is reached")
+        void shouldOpenImmediatelyWhenMinimumCountReachedAndRateIsZero() {
```
Comment (ggivo) [inline src/test/java/io/lettuce/core/failover/CircuitBreakerUnitTests.java]: don't see what we test in more compared to shouldConsiderOnlyCountWhenFailureRateThresholdIsZero()

**Comment by atakavci** [comment]
> What about providing
> CircuitBreaker.recordFailure()
> CircuitBreaker.recordSuccess()

@ggivo good point, doing it right now.

File: src/test/java/io/lettuce/core/failover/CircuitBreakerUnitTests.java (hunk 1 above)
Comment (atakavci) [reply src/test/java/io/lettuce/core/failover/CircuitBreakerUnitTests.java]: dropping this one `shouldOpenImmediatelyWhenMinimumCountReachedAndRateIsZero`
